controllers/hibp.js

```diff
@@ -4,35 +4,52 @@ const AppConstants = require("../app-constants");
 const DB = require("../db/DB");
 const EmailUtils = require("../email-utils");
 const HIBP = require("../hibp");
-
+const sha1 = require("../sha1-utils");
+const HBSHelpers = require("../hbs-helpers");
 
 async function notify (req, res) {
+
+  const hashes = req.body.hashSuffixes.map(suffix=>req.body.hashPrefix + suffix);
+  const subscribers = await DB.getSubscribersByHashes(hashes);
+
   const reqBreachName = req.body.breachName.toLowerCase();
-  let breach = HIBP.getBreachByName(req.app.locals.breaches, reqBreachName);
-  if (!breach) {
+  let breachAlert = HIBP.getBreachByName(req.app.locals.breaches, reqBreachName);
+  
+  if (!breachAlert) {
     // If breach isn't found, try to reload breaches from HIBP
     await HIBP.loadBreachesIntoApp(req.app);
-    breach = HIBP.getBreachByName(req.app.locals.breaches, reqBreachName);
-    if (!breach) {
+    breachAlert = HIBP.getBreachByName(req.app.locals.breaches, reqBreachName);
+    if (!breachAlert) {
       // If breach *still* isn't found, we have a real error
       throw new Error("Unrecognized breach: " + reqBreachName);
     }
   }
 
-  const hashes = req.body.hashSuffixes.map(suffix=>req.body.hashPrefix + suffix);
-  const subscribers = await DB.getSubscribersByHashes(hashes);
+  console.log(`Found ${subscribers.length} subscribers in ${breachAlert.Name}. Notifying ...`);
 
-  console.log(`Found ${subscribers.length} subscribers in ${breach.Name}. Notifying ...`);
   const notifiedSubscribers = [];
+
   for (const subscriber of subscribers) {
     const email = subscriber.email;
+
+    const unsafeBreachesForEmail = await HIBP.getUnsafeBreachesForEmail(
+      sha1(email),
+      req.app.locals.breaches
+    );
+
     if (!notifiedSubscribers.includes(email)) {
       console.log(email);
```

Comment:
Unrelated, but is this cool? Does this raw email address show up in server logs or anything? /cc @groovecoder

Reply:
Oh, yeah we don't want to log any plaintext email addresses.